gnosis/safe-react
# Editable Prefixed Address Input EIP-3770

## What it solves
Resolves #2756

## How this PR fixes it

Added network prefix validations in `AddressInput` and `AddressBookInput` components.

Added unit tests to `AddressInput` and `AddressBookInput` components, and `getAddressWithoutNetworkPrefix` and `getNetworkPrefix` util functions

## How to test it

Places where the Address input appears:

1. Load Safe Form (Step two, in the load safe address input).
2. Create Safe Form (Step three, Owners and Confirmations, in the Owners address inputs).
3. Settings > Owners section, Add & Replace owners Dialogs, in the Owners address inputs.
4. AddressBook Section > Create Entry Dialog
5. New Transaction > Send Founds Dialog, Recipient input.
6. New Transaction > Send collectible Dialog, Recipient input.
7. New Transaction > Contract Interaction, Contract Address input.

## Use cases

1. Use a valid network prefix in a address input (rin:0xC245cb45B044d66fbE8Fb33C26c0b28B4fc367B2)
2. Network prefixes that don't match with the current network show an error ("The current network doesn't match the given address")
3. Invalid network prefix shows an error ("Unsupported network prefix")
4. Checksum address with a valid network prefix (rin:0X9913B9180C20C6B0F21B6480C84422F6EBC4B808 -> rin:0x9913B9180C20C6b0F21B6480c84422F6ebc4B808)
5. Load an address with a QR code should work.
6. Load an address with ENS name  should work.
7. All the validations like "duplicated owner address" should work with prefixed address



## Screenshots
![Captura de pantalla 2021-11-03 a las 10 37 22](https://user-images.githubusercontent.com/26763673/140308469-7c1c015d-bd7a-42e5-a71d-b3e87cb3c319.png)
![Captura de pantalla 2021-11-04 a las 12 42 10](https://user-images.githubusercontent.com/26763673/140308474-1c2fed57-e82f-4be7-b73b-66eabbf54d54.png)
![Captura de pantalla 2021-11-04 a las 12 42 32](https://user-images.githubusercontent.com/26763673/140308476-06b93f92-3f19-4f7b-9635-5151435598c3.png)
![Captura de pantalla 2021-11-04 a las 12 42 51](https://user-images.githubusercontent.com/26763673/140308479-f47d849f-0b42-43b9-a325-ed2d4e1896c0.png)
![Captura de pantalla 2021-11-04 a las 12 45 01](https://user-images.githubusercontent.com/26763673/140308483-ebfb8738-12bf-477b-81a3-6ca41bdf314c.png)
![Captura de pantalla 2021-11-04 a las 12 46 03](https://user-images.githubusercontent.com/26763673/140308484-67ed90e6-e0b0-4a0c-bb0f-9f8d370ac1c6.png)
![Captura de pantalla 2021-11-04 a las 13 16 49](https://user-images.githubusercontent.com/26763673/140311953-04bfce53-85d6-4cc9-bf52-472b7121cd0d.png)
